File: app/src/main/java/net/numa08/gochisou/presentation/internal/di/modules/ApplicationModule.java
package net.numa08.gochisou.presentation.internal.di.modules;

import android.content.Context;

import net.numa08.gochisou.GochisouApplication;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
@Singleton
public class ApplicationModule {

    private final GochisouApplication application;

    public ApplicationModule(GochisouApplication application) {
        this.application = application;
    }

    @Provides
    public Context providesContext() {
        return application;
    }
}

File: app/src/main/java/net/numa08/gochisou/presentation/internal/di/modules/ApplicationModule.kt
package net.numa08.gochisou.presentation.internal.di.modules

import android.content.Context
import android.content.SharedPreferences
import dagger.Module
import dagger.Provides
import org.jetbrains.anko.defaultSharedPreferences
import javax.inject.Singleton

@Module
@Singleton
class ApplicationModule(val application: Context) {

    @Provides
    @Singleton
    fun providesContext(): Context {
        return application
    }

    @Provides
    @Singleton
    fun providesSharedPreferences(context: Context) : SharedPreferences = context.defaultSharedPreferences
}
